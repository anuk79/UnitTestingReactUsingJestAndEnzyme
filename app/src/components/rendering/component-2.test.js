
// vendors
import React from "react";
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";

// components
import ComponentTwo from './component-2';

describe("Component Two", () => {
    it("should render correctly", () => {
        const wrapper = shallow(<ComponentTwo />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('input').exists()).toBeTruthy();
        expect(wrapper.find('.error-message').exists()).toBeFalsy();
    });

    it("should render correctly when error in props", () => {
        const wrapper = shallow(<ComponentTwo 
            username='test user name' 
            isError={true} 
            errorMessage='test error message' 
        />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('input').exists()).toBeTruthy();
        expect(wrapper.find('.error-message').exists()).toBeTruthy();
    });

    it('should call handleChange of props when input value changes', () => {
        const handleChangeMock = jest.fn();
        const wrapper = shallow(<ComponentTwo handleChange={handleChangeMock}/>);
        wrapper.find('input').simulate('change')
        expect(handleChangeMock).toBeCalled();
    });
});