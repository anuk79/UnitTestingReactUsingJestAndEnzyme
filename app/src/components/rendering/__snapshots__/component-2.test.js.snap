// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Two should render correctly 1`] = `
<Fragment>
  <input
    onChange={[Function]}
    type="text"
  />
</Fragment>
`;

exports[`Component Two should render correctly when error in props 1`] = `
<Fragment>
  <input
    onChange={[Function]}
    type="text"
    value="test user name"
  />
  <span
    className="error-message"
  >
    test error message
  </span>
</Fragment>
`;
